R/models.R

```diff
@@ -104,7 +104,11 @@ predict_model.xgb.Booster <- function(x, newdata) {
   # Test model type
   model_type <- model_type(x)
 
-  predict(x, as.matrix(newdata))
+  if (ncol(newdata) > length(x$feature_names)) {
+    predict(x, as.matrix(newdata)[, x$feature_names, drop = FALSE])
```

Comment:
I guess that always running this line (i.e. remove the if-else test would fix my issue with correct features and wrong order, but I generally think it is better than handle that outside predict_model, and rather ensure that we always provide exactly the columns that we need in the right order, to these predict_model functions.

Reply:
Should be fixed now.